### PR TITLE
Product Family detail page - specification attributes

### PR DESCRIPTION
**Pull request by sharanagoudapatil**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.page/
- After: https://643-product-spec-attributes--danaher-ls-aem--hlxsites.hlx.page/us/en/products/family/biomek-i7-automated-liquid-handlers
- After: https://643-product-spec-attributes--danaher-ls-aem--hlxsites.hlx.page/us/en/products/sku/548021-molecular-devices

### CODE REVIEW

File: blocks/product-specifications/product-specifications.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { div, h1 } from '../../scripts/dom-builder.js';
+
+export default function decorate(block) {
+  let response;
+  if (localStorage.getItem('product-hero')) {
```
Comment (duynguyen) [inline blocks/product-specifications/product-specifications.js:5]: let's use the common script `getProductResponse()` that Ravi added in #640

**Review by davenichols-DHLS — APPROVED** [review]
Please see Duy's comments, but I approve from the UX point of view, the tables are looking good.

File: blocks/product-specifications/product-specifications.js (hunk 1 above)
Comment (sharanagoudapatil) [reply blocks/product-specifications/product-specifications.js]: @duynguyen - I've modified as per above comments. Please have a look. Thanks.

**Comment by davidzochowski** [comment]
@sharanagoudapatil I tried to test with another page but it still renders the biomek family.
https://643-product-spec-attributes--danaher-ls-aem--hlxsites.hlx.page/us/en/products/sku/7181375-molecular-devices

File: blocks/product-specifications/product-specifications.js
Comment (davidzochowski) [inline blocks/product-specifications/product-specifications.js:1]: As part of the attributejson, I added another field called "unit". Would you please append the unit (assuming non-empty/non-null) to the value in the table?

**Review by davidzochowski — DISMISSED** [review]
As part of the attributejson, I added another field called "unit". Would you please append the unit (assuming non-empty/non-null) to the value in the table?

Edit: For the attribute values that come in a list, could you add a space between the values? For the example page you used, instead of T1020_WB,T1070,T190F_LLS,T230,T230_LSS,... , could it be T1020_WB, T1070, T190F_LLS, T230, T230_LSS, ... (there are spaces after each comma)

**Comment by sharanagoudapatil** [comment]
@davidzochowski - Please review the changes as requested. Thanks.

File: blocks/product-specifications/product-specifications.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { div, h1 } from '../../scripts/dom-builder.js';
+import { getProductResponse } from '../../scripts/scripts.js';
+
+export default function decorate(block) {
+  const response = getProductResponse();
+  if (response?.length > 0) {
+    const attrContainer = div({ class: 'block' });
+    if (response[0]?.raw.attributejson !== undefined) {
+      const attrJson = JSON.parse(response[0]?.raw.attributejson);
+      if (attrJson.length > 0) {
```
Comment (duynguyen) [inline blocks/product-specifications/product-specifications.js:10]: this condition might be redundant

File: blocks/product-specifications/product-specifications.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        attrJson.forEach((item) => {
+          const attrWrapper = div({ class: 'attr-wrapper' });
+          const tableCaption = div(
+            { class: 'sm:flex sm:items-center pt-12' },
+            div(
+              { class: 'sm:flex-auto' },
+              h1({ class: 'text-xl font-normal leading-6 text-black' }, item.label),
```
Comment (duynguyen) [inline blocks/product-specifications/product-specifications.js:17]: do we want an h1 here? then there would be multiple h1 on the page, and the a11y score would be affect. let's revise the heading level here.

File: blocks/product-specifications/product-specifications.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            ),
+          );
+          const tableContainer = div({ class: 'min-w-full mt-2 border divide-y divide-gray-300 rounded-lg' });
+          if (item.value.length) {
```
Comment (duynguyen) [inline blocks/product-specifications/product-specifications.js:21]: redundant condition

File: blocks/product-specifications/product-specifications.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+              );
+              tableContainer.append(tableRow);
+            });
+          }
+          attrWrapper.append(tableCaption);
+          attrWrapper.append(tableContainer);
+          attrContainer.append(attrWrapper);
+        });
+      }
+      block.innerHTML = '';
```
Comment (duynguyen) [inline blocks/product-specifications/product-specifications.js:55]: if these 2 lines are common for both conditions, let's move it outside of the if and remove it in the else.

**Review by duynguyen — CHANGES_REQUESTED** [review]
thanks @sharanagoudapatil , I left some comments.

**Comment by davidzochowski** [comment]
@sharanagoudapatil I was thinking something like this.
Screenshot 2024-01-11 at 9 31 17 AM

**Comment by sharanagoudapatil** [comment]
@davidzochowski - Done with the changes. Pls check. Thanks.

File: blocks/product-specifications/product-specifications.js (hunk 3 above)
Comment (sharanagoudapatil) [reply blocks/product-specifications/product-specifications.js]: Hi @duynguyen - I've made the changes for rest of the comments. But will connect for this H1 tag as I'm not getting your comments here. Thanks.

Comment (sharanagoudapatil) [reply blocks/product-specifications/product-specifications.js]: As as discussed with @davenichols-DHLS I've made the table headings from h1 to h2 with the same css changes. Please have a look. Thanks.

File: blocks/product-specifications/product-specifications.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import { div, h2 } from '../../scripts/dom-builder.js';
+import { getProductResponse } from '../../scripts/scripts.js';
+
+export default function decorate(block) {
+  const response = getProductResponse();
+  if (response?.length > 0) {
+    const attrContainer = div({ class: 'block' });
```
Comment (duynguyen) [inline blocks/product-specifications/product-specifications.js:7]: let's remove the wrapper div as it's not needed

**Review by duynguyen — APPROVED** [review]
minor comment. good with me to merge.

File: blocks/product-specifications/product-specifications.js (hunk 6 above)
Comment (sharanagoudapatil) [reply blocks/product-specifications/product-specifications.js]: Removed.

**Comment by davidzochowski** [comment]
@sharanagoudapatil Looks good to me. Approved

**Review by davidzochowski — APPROVED** [review]

**Pull request merged**